### PR TITLE
include "pip install -e ." in setup steps

### PR DESCRIPTION
**Pull request by landonr**
include "pip install -e ." in setup steps

## Description:
This was required for me to run the codeowners script

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [ ] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: guides/contributing.rst [hunk 1]
```diff
@@ -337,6 +337,7 @@ This is only possible for ``pip`` installs.
     # Install ESPHome
     cd esphome/
     script/setup
+    pip install -e .
```
Comment (guillempages) [inline guides/contributing.rst:340]: Maybe the pip install should be before the setup call?
```suggestion
    pip install -e .
    script/setup
```

**Review by jesserockz — COMMENTED** [review]
This is already inside `script/setup`
https://github.com/esphome/esphome/blob/b20bae23ccac3903239ca68bfb3a4add25b7b44d/script/setup#L20

**Comment by landonr** [comment]
> This is already inside `script/setup` https://github.com/esphome/esphome/blob/b20bae23ccac3903239ca68bfb3a4add25b7b44d/script/setup#L20

on mac with pip installed, I had to run this separately

**Comment by landonr** [comment]
@jesserockz should I just decline this? Or is there a better way to add this?

**Pull request closed without merging**